scripts/categorizer.py

```diff
@@ -74,24 +71,172 @@ def strip_comments_after_code(lines, comment_lines):
         if comment_lines[line_number] == 1:
             continue
 
-        # Try to find if a comment exists on code line
-        try:
-            comment_line_index = line.index("#")
-        except ValueError:
-            comment_line_index = None
+        # This will be overwritten in while loop pretty quickly
+        comment_symbol_index = 0
+
+        # copy of the line
+        string_to_search = line
+        # Figuring out whether there is a comment char not in a string
+        while comment_symbol_index != None:
+
+            # Try to find if a comment exists on code line
+            try:
+                comment_symbol_index = string_to_search.index("#")
+            except ValueError:
+                comment_symbol_index = None
+
+            if comment_symbol_index != None:
+
+                if not check_if_in_string(line, comment_symbol_index):
```

Comment:
Rename function to is_index_in_string.

Reply:
I want to test conditionals with this function name, so it will stay the same for now...